pandas/core/arrays/string_arrow.py

```diff
@@ -0,0 +1,619 @@
+        self, dtype=None, copy: bool = False, na_value=lib.no_default
+    ) -> np.ndarray:
+        """
+        Convert to a NumPy ndarray.
+        """
+        # TODO: copy and na_value arguments are ignored
+        return self.data.__array__(dtype=dtype)
+
+    def __len__(self) -> int:
+        """
+        Length of this array.
+
+        Returns
+        -------
+        length : int
+        """
+        return len(self.data)
+
+    @classmethod
+    def _from_sequence_of_strings(cls, strings, dtype=None, copy=False):
+        return cls._from_sequence(strings, dtype=dtype, copy=copy)
+
+    @classmethod
+    def _from_factorized(cls, values, original):
+        return cls._from_sequence(values)
+
+    @classmethod
+    def _concat_same_type(cls, to_concat) -> ArrowStringArray:
+        """
+        Concatenate multiple ArrowStringArray.
+
+        Parameters
+        ----------
+        to_concat : sequence of ArrowStringArray
+
+        Returns
+        -------
+        ArrowStringArray
+        """
+        return cls(
+            pa.chunked_array(
+                [array for ea in to_concat for array in ea.data.iterchunks()]
+            )
+        )
+
+    def __getitem__(self, item: Any) -> Any:
+        """Select a subset of self.
+
+        Parameters
+        ----------
+        item : int, slice, or ndarray
+            * int: The position in 'self' to get.
+            * slice: A slice object, where 'start', 'stop', and 'step' are
+              integers or None
+            * ndarray: A 1-d boolean NumPy ndarray the same length as 'self'
+
+        Returns
+        -------
+        item : scalar or ExtensionArray
+
+        Notes
+        -----
+        For scalar ``item``, return a scalar value suitable for the array's
+        type. This should be an instance of ``self.dtype.type``.
+        For slice ``key``, return an instance of ``ExtensionArray``, even
+        if the slice is length 0 or 1.
+        For a boolean mask, return an instance of ``ExtensionArray``, filtered
+        to the values where ``item`` is True.
+        """
+        item = check_array_indexer(self, item)
+
+        if isinstance(item, abc.Iterable):
+            if not is_array_like(item):
+                item = np.array(item)
+            if len(item) == 0:
+                return type(self)(pa.chunked_array([], type=pa.string()))
+            elif is_integer_dtype(item):
+                return self.take(item)
+            elif is_bool_dtype(item):
+                return type(self)(self.data.filter(item))
+            else:
+                raise IndexError(
+                    "Only integers, slices and integer or "
+                    "boolean arrays are valid indices."
+                )
+
+        value = self.data[item]
+        if isinstance(value, pa.ChunkedArray):
+            return type(self)(value)
+        else:
+            return self._as_pandas_scalar(value)
+
+    def _as_pandas_scalar(self, arrow_scalar: pa.Scalar):
```

Comment:
type output

Reply:
see also https://github.com/pandas-dev/pandas/pull/35259#discussion_r517352902

Reply:
`_as_pandas_scalar` is an artefact of fletcher. It was a module level function and was moved to a class method to access self._dtype.na_value in https://github.com/pandas-dev/pandas/pull/35259/commits/1a200a2b00dc25e57dce1a7f59fa40ab6fb1ae27 see https://github.com/pandas-dev/pandas/pull/35259#discussion_r517351810

as stated in https://github.com/pandas-dev/pandas/pull/35259/files#r522951940 the return type of \_\_getitem__ is either the same as the return type here or an array of that type.

`_as_pandas_scalar` is only used in \_\_getitem__ . I'm not sure if we will want to re-use this. maybe in a base class for arrow backed EA. or we could subsume this method into \_\_getitem__

Reply:
I think there will come more use-cases of `_as_pandas_scalar`. There are other functions that can return a scalar as well, apart from `__getitem__`, such as reductions (although for string data specifically, there are not many reductions supported of course)

Reply:
sure. am happy keeping it. will probably create an alias for the return type of `_as_pandas_scalar` and then use in the return type of \_\_getitem__.  have opened #37898 as a precursor for the discussion on typing of \_\_getitem__ in EA. 

see https://github.com/pandas-dev/pandas/pull/37898#issue-521850554

> could create a scalar type alias in each of the EA classes similar to DTScalarOrNaT in DatetimeLikeArrayMixin

Reply:
yeah would do this more generally with support for scalars in the base class as a first class item